galley/cmd/gals/cmd/validator.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+
+	"istio.io/istio/galley/cmd/shared"
+	"istio.io/istio/galley/pkg/crd/validation"
+	"istio.io/istio/pilot/pkg/bootstrap"
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/util"
+)
+
+// waitSignal awaits for SIGINT or SIGTERM and closes the channel
+func waitSignal(stop chan struct{}) {
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	<-sigs
+	close(stop)
+	_ = log.Sync()
+}
+
+func validatorCmd(printf, fatalf shared.FormatFn) *cobra.Command {
+	var (
+		webhookConfigName   string
+		pilotWebhookName    string
+		mixerWebhookName    string
+		port                uint
+		domainSuffix        string
+		certFile            string
+		keyFile             string
+		caFile              string
+		healthCheckInterval time.Duration
+		healthCheckFile     string
+	)
+
+	patchCert := func(stop chan struct{}) error {
+		caCertPem, err := ioutil.ReadFile(caFile)
+		if err != nil {
+			return err
+		}
+
+		timeout := time.After(time.Minute)
+		cl := common.client.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations()
+		for {
+			webhooks := []string{pilotWebhookName, mixerWebhookName}
+			if err = util.PatchValidatingWebhookConfig(cl, webhookConfigName, webhooks, caCertPem); err == nil {
+				return nil
+			}
+
+			log.Errorf("Patching caBundle in ValidatingWebhookConfigurations %v failed: %v -- retrying",
+				webhookConfigName, err)
+			select {
+			case <-stop:
+				// canceled by caller
```

Comment:
Nit : cancelled

Reply:
https://www.grammarly.com/blog/canceled-vs-cancelled. Given my emacs dictionary is american I chose "cancel". :)